### PR TITLE
add L3,  Jaccard, Hamming  evaluator

### PR DESCRIPTION
**Pull request by smy0428**
Add L3 norm (Minkowski distance where p=3) for embedding evaluator.
I know it is a bit pointless to write L3 norm than to L1, L2 or L infinity, but all the other ordinary norms have been written. Minkowski distance can also be extended to any p >= 1, just more math problems. 
Plus Jaccard similarity evaluator and Hamming distance evaluator for text.

### CODE REVIEW

File: evaluators/embedding/L3NormEvaluator/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# L1NormEvaluator
```
Comment (cristianmtr) [inline evaluators/embedding/L3NormEvaluator/README.md:1]: I'm confused. Is it L1 or L3? the folder is called 'L3...', but there are references to L1 throughout

Comment (smy0428) [reply evaluators/embedding/L3NormEvaluator/README.md]: Sorry, I forgot to change them. Now should be fixed.

**Comment by JoanFM** [comment]
Hello @smy0428 ,

it would be good if you could open one PR for `executor`. Like this we could review one by one

**Review by JoanFM — CHANGES_REQUESTED** [review]
Thank you very much for the contribution, but having independent PRs would really ease the process of reviewing

File: evaluators/text/JaccardSimilarityEvaluator/__init__.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    @property
+    def metric(self):
+        return 'JaccardSimilarity'
+
+    def evaluate(self, actual: str, desired: str):
+        a = set(actual.lower().split())
```
Comment (JoanFM) [inline evaluators/text/JaccardSimilarityEvaluator/__init__.py:15]: please use better variable naming

File: evaluators/text/JaccardSimilarityEvaluator/__init__.py [hunk 3]
```diff
@@ -12,9 +12,9 @@ def metric(self):
...
-        a = set(actual.lower().split())
-        b = set(desired.lower().split())
-        c = a.intersection(b)
-        total = len(a) + len(b) - len(c)
-        return float(len(c)) / total if total != 0.0 else 0.0
+        acutal_words = set(actual.lower().split())
+        desired_words = set(desired.lower().split())
+        intersection = acutal_words.intersection(desired_words)
+        union_length = len(acutal_words) + len(desired_words) - len(intersection)
+        return float(len(intersection)) / union_length if union_length != 0.0 else 0.0
```
Comment (JoanFM) [inline evaluators/text/JaccardSimilarityEvaluator/__init__.py:19]: why is this `float` conversion required?

**Review by JoanFM — CHANGES_REQUESTED** [review]
Please, remember to split the PR

File: evaluators/text/JaccardSimilarityEvaluator/__init__.py (hunk 3 above)
Comment (smy0428) [reply evaluators/text/JaccardSimilarityEvaluator/__init__.py:19]: You are right, it is not necessary in python.

**Comment by nan-wang** [comment]
Bad commit message, please refer to:
- [Commit Message Guideline for the First Time Contributor](https://github.com/jina-ai/jina/issues/553)
- [Contributing Guideline](https://github.com/jina-ai/jina/blob/master/CONTRIBUTING.md)
#553 :bookmark: Commit Message Guideline for the First Time Contributor
jina-ai/jina | Jun 21st | Added by GitHub
CONTRIBUTING.md
jina-ai/jina | Added by GitHub

**Review by nan-wang — COMMENTED** [review]
hi, there. I'm wondering is there any special reason why scipy is not used for calculating these distances?
reference: https://docs.scipy.org/doc/scipy/reference/spatial.distance.html

**Comment by JoanFM** [comment]
Closing since it has been divided in 3 different PRs

**Pull request closed without merging**